hr_holidays_leave_auto_approve/models/hr_holidays.py

```diff
@@ -0,0 +1,17 @@
+# -*- coding: utf-8 -*-
+# © 2016 ONESTEiN BV (<http://www.onestein.eu>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openerp import models, api
+
+
+class HrHolidays(models.Model):
+    _inherit = "hr.holidays"
+
+    @api.model
+    @api.returns('self', lambda value: value.id)
+    def create(self, values):
+        result = super(HrHolidays, self).create(values)
+        if result.holiday_status_id.auto_approve:
+            result.sudo().signal_workflow('validate')
```

Comment:
When the admin user doesn't have the `base.group_hr_user` group, automatic approve doesn't work : the workflow raises the "Contact a human resource manager" error.

I think it would be better to :
- Add a new workflow transition from `confirm` to `validate`, with the `base.group_user` group (employee), `holiday_status_id.auto_approve` as condition, and no signal.
- Override the `_check_state_access_right`, to allow changing the state to `validate` if the defined `holiday_status_id` allows auto approve. You don't have the record in this method, but when called from create, you will always have the `holiday_status_id` field.

By doing this :
- You won't need sudo (group employee) : No access rights issue, consistent `write_uid`, etc.
- You won't need to call signal_workflow (no signal) : No need to override the create method.